pygmt/helpers/utils.py

```diff
@@ -61,23 +66,41 @@ def data_kind(data, x=None, y=None, z=None, required_z=False):
     'file'
     >>> data_kind(data=pathlib.Path("my-data-file.txt"), x=None, y=None)
     'file'
+    >>> data_kind(data=None, x=None, y=None, optional_data=True)
+    'null'
+    >>> data_kind(data=2.0, x=None, y=None, optional_data=True)
+    'null'
+    >>> data_kind(data=True, x=None, y=None, optional_data=True)
+    'null'
```

Comment:
Somewhat off-topic, but thinking about whether we need a doctest for the `required_z` parameter here instead of just at https://github.com/GenericMappingTools/pygmt/blob/59a203579a2b896b0d721fa88b65d7b97a5115f3/pygmt/tests/test_clib.py#L427-L453

Reply:
Done in 73c31e214